test/connectivity/connectivity.spec.js

```diff
@@ -0,0 +1,305 @@
+      }
+    },
+    shell: {}
+  }
+};
+m._load = function hookedLoader(request, parent, isMain) {
+  const stub = stubs[request];
+  return stub || originalLoader(request, parent, isMain);
+};
+
+// Mock dom/window for tests.
+// Has to come before @mongodb-js/compass-connect require.
+jsdomGlobal();
+
+const AppRegistry = require('hadron-app-registry');
+const Connection = require('mongodb-connection-model');
+const CompassConnectPlugin = require('@mongodb-js/compass-connect');
+const CompassConnectComponent = CompassConnectPlugin.default;
+const activateCompassConnect = CompassConnectPlugin.activate;
+
+const deactivateCompassConnect = CompassConnectPlugin.deactivate;
+
+const {
+  dockerComposeDown,
+  dockerComposeUp
+} = require('./docker-instance-manager');
+
+// Hide react warnings.
+// eslint-disable-next-line no-console
+const originalWarn = console.warn.bind(console.warn);
+// eslint-disable-next-line no-console
+console.warn = (msg) => (
+  !msg.toString().includes('componentWillReceiveProps')
+  && !msg.toString().includes('componentWillUpdate')
+  && originalWarn(msg)
+);
+
+const delay = promisify(setTimeout);
+
+const connectionsToTest = [{
+  title: 'local community standalone',
+  dockerComposeFilePath: 'community/docker-compose.yaml',
+  model: {
+    hostname: 'localhost',
+    port: 27020,
+  },
+  expectedInstanceDetails: {
+    _id: 'localhost:27020',
+    authenticatedUsers: [],
+    build: {
+      raw: {
+        version: /4.4.[1-9]+$/
+      }
+    },
+    client: {
+      isWritable: true,
+      isMongos: false,
+    },
+    dataLake: { isDataLake: false, version: null },
+    genuineMongoDB: { isGenuine: true, dbType: 'mongodb' },
+    host: {
+      os: 'Ubuntu',
+      os_family: 'linux',
+      kernel_version: '18.04',
+      arch: 'x86_64'
+    },
+    hostname: 'localhost',
+    port: 27020
+  }
+}];
+
+describe('Connectivity', () => {
+  connectionsToTest.forEach(({
+    dockerComposeFilePath,
+    expectedInstanceDetails,
+    model: testConnectionModel,
+    title: testConnectionTitle
+  }) => {
+    let wrapper;
+    let appRegistry;
+    let compassConnectStore;
+
+    before(async() => {
+      dockerComposeUp(dockerComposeFilePath);
+
+      appRegistry = new AppRegistry();
+
+      activateCompassConnect(appRegistry);
+
+      global.hadronApp = {
+        appRegistry
+      };
+
+      const ROLE = {
+        name: 'Status',
+        component: () => (<div id="statusPlugin">Status</div>)
+      };
+      global.hadronApp.appRegistry = appRegistry;
+      global.hadronApp.appRegistry.registerRole('Application.Status', ROLE);
+
+      compassConnectStore = appRegistry.getStore('Connect.Store');
+
+      // Remove all logic around saving and loading stored connections.
+      // NOTE: This is tightly coupled with the store in compass-connect.
+      // https://github.com/mongodb-js/compass-connect/blob/master/src/stores/index.js
+      compassConnectStore._saveRecent = () => {};
```

Comment:
I'd still put all of this mocking somewhere else.

Reply:
Seeing how we can avoid stubbing them altogether